linode/manager
# M3-1770 Group Import CTA

# Group Import CTA

## Description

This PR adds a CTA on the Dashboard to import Groups and convert them to tags.

<img width="608" alt="screen shot 2018-12-13 at 9 50 03 am" src="https://user-images.githubusercontent.com/16911484/49946850-d1fb9500-febd-11e8-93ba-c146a50a3cd6.png">

Included is a memoized selector (using Reselect) needed to get Linodes with eligible groups to import. This selector can be reused in the Group Import Drawer.

## To test

1) Have a Linode with a display group (via Classic Manager) without a corresponding tag.
2) Go to the Dashboard. 
3) **Observe:** the CTA will be displayed.
4) Add the name of the display group as a tag for your Linode.
5) Refresh the application. 
6) Go to the Dashboard.
7) **Observe:** the CTA will NOT be displayed.

## Notes:
Step 5 (refreshing the application) is necessary because the selector is memoized, meaning it won't re-run unless there is a change to the appropriate piece of Redux state (`__resources.linodes.entities`). When adding/deleting tags are refactored as events updating Redux state, this step won't be necessary.

#### New Library:
[Reselect](https://github.com/reduxjs/reselect) (it's awesome). 

#### New Component:
`src/features/Dashboard/GroupImportCard`

#### New directory:
`src/store/selectors` <-- where Reselect selectors will live
